### PR TITLE
Add SSEFaultHandler unit coverage

### LINKED ISSUE #183: Add unit tests for SSEFaultHandler
## Summary

The `SSEFaultHandler` class has no unit or integration tests. It implements Server-Sent Events (SSE) for real-time fault streaming and is the most complex handler to test due to its streaming, multi-client, and reconnection logic.

**Methods/features to test:**
- `handle_stream` - establishes SSE connection, sends initial faults, then streams updates
- `on_fault_event` - callback that broadcasts fault events to all connected SSE clients
- Multi-client support (concurrent viewers)
- Keepalive/heartbeat mechanism
- Client reconnection with `Last-Event-ID` header
- Graceful disconnection handling

---

## Proposed solution (optional)

Create `test/test_sse_fault_handler.cpp`:

```cpp
#include <gtest/gtest.h>
#include "ros2_medkit_gateway/http/handlers/sse_fault_handler.hpp"

class SSEFaultHandlerTest : public ::testing::Test {};

// Test: SSE response has correct Content-Type header
TEST_F(SSEFaultHandlerTest, StreamSetsCorrectContentType) {
  // Verify: Content-Type: text/event-stream
}

// Test: SSE event format follows spec (data: ...\n\n)
TEST_F(SSEFaultHandlerTest, EventFormatFollowsSSESpec) {
  // Verify: "id: ...\nevent: fault\ndata: {...}\n\n"
}

// Test: Reconnection with Last-Event-ID replays missed events
TEST_F(SSEFaultHandlerTest, ReconnectionReplaysFromLastEventId) { /* ... */ }
```

Alternatively, this could be an **integration test** in Python (`test_sse_faults.test.py`) using `sseclient` or `httpx-sse`:

```python
def test_sse_stream_receives_fault_events(self):
    with httpx.stream("GET", f"{BASE_URL}/faults/stream") as response:
        assert response.headers["content-type"] == "text/event-stream"
        # Inject a fault, verify event received
```

### PR DESCRIPTION
**Pull request by eclipse0922**
# Pull Request

## Summary

Add dedicated `SSEFaultHandler` unit coverage for stream setup, replay behavior, multi-client handling, disconnect cleanup, and keepalive heartbeats. To make heartbeat behavior testable without waiting 30 real seconds, the handler now accepts an internal keepalive interval override while keeping the production default unchanged.

---

## Issue

Link the related issue (required):

- closes #183

---

## Type

- [ ] Bug fix
- [x] New feature or tests
- [ ] Breaking change
- [ ] Documentation only

---

## Testing

How was this tested / how should reviewers verify it?

- `colcon test --packages-select ros2_medkit_gateway --event-handlers console_direct+ --ctest-args -R test_sse_fault_handler -V` in `osrf/ros:jazzy-desktop`
- `./scripts/test.sh lint --packages-select ros2_medkit_gateway` in `osrf/ros:jazzy-desktop`

---

## Checklist

- [x] Breaking changes are clearly described (and announced in docs / changelog if needed)
- [x] Tests were added or updated if needed
- [x] Docs were updated if behavior or public API changed

### CODE REVIEW

File: src/ros2_medkit_gateway/CMakeLists.txt [hunk 1]
```diff
@@ -471,6 +471,12 @@ if(BUILD_TESTING)
   ament_add_gtest(test_sse_transport_provider test/test_sse_transport_provider.cpp)
   target_link_libraries(test_sse_transport_provider gateway_lib)
 
+  # Add SSE fault handler tests
+  ament_add_gtest(test_sse_fault_handler test/test_sse_fault_handler.cpp)
+  target_link_libraries(test_sse_fault_handler gateway_lib)
```
Comment (bburda) [inline src/ros2_medkit_gateway/CMakeLists.txt:476]: **[Blocking]** `test_sse_fault_handler` is not in the `_test_targets` list (lines 635-692) for `ENABLE_COVERAGE`. Every other gateway test target is listed there. Without this, coverage builds will not instrument the new test binary, so the SSE fault handler lines exercised here will not appear in Codecov reports - undermining the stated purpose of this PR.

Add `test_sse_fault_handler` to `_test_targets` after `test_sse_transport_provider` (line 662).

File: src/ros2_medkit_gateway/test/test_sse_fault_handler.cpp [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+TEST_F(SSEFaultHandlerTest, StreamReplaysBufferedEventsUsingSseFormat) {
+  enqueue_event(make_fault_event(FaultEvent::EVENT_CONFIRMED, "TEMP_HIGH", 123, 456000000));
+
+  auto req = make_stream_request("127.0.0.1");
+  httplib::Response res;
+  handler_->handle_stream(req, res);
+
+  auto output = read_stream_once(res, 1);
+
```
Comment (bburda) [inline src/ros2_medkit_gateway/test/test_sse_fault_handler.cpp:281]: **[Blocking]** Missing edge case test: invalid `Last-Event-ID` header.

`handle_stream` catches `std::stoull` exceptions for invalid `Last-Event-ID` values and falls back to `last_event_id = 0` (full buffer replay). This exception-handler branch has zero test coverage.

Suggested test:
```cpp
TEST_F(SSEFaultHandlerTest, InvalidLastEventIdFallsBackToFullReplay) {
  enqueue_event(make_fault_event(FaultEvent::EVENT_CONFIRMED, "FAULT_X", 200));

  auto req = make_stream_request("127.0.0.1", "not-a-number");
  httplib::Response res;
  handler_->handle_stream(req, res);

  auto output = read_stream_once(res, 1);
  EXPECT_NE(output.find("id: 1\n"), std::string::npos);
  EXPECT_EQ(parse_sse_payload(output)["fault"]["fault_code"], "FAULT_X");

  release_stream(res);
}
```

File: src/ros2_medkit_gateway/test/test_sse_fault_handler.cpp [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+  auto req = make_stream_request("127.0.0.1");
+  httplib::Response res;
+
+  handler_->handle_stream(req, res);
+  EXPECT_EQ(handler_->connected_clients(), 1u);
+
+  release_stream(res, false);
+
+  EXPECT_EQ(handler_->connected_clients(), 0u);
+}
```
Comment (bburda) [inline src/ros2_medkit_gateway/test/test_sse_fault_handler.cpp:425]: **[Blocking]** Missing edge case test: buffer overflow / FIFO eviction.

`on_fault_event` evicts the oldest entry when the buffer exceeds `kMaxBufferedEvents = 100`. This eviction path is completely untested. An off-by-one bug or wrong pop order would be invisible.

Suggested test:
```cpp
TEST_F(SSEFaultHandlerTest, BufferEvictsOldestEventWhenFull) {
  for (int i = 1; i <= 101; ++i) {
    enqueue_event(make_fault_event(FaultEvent::EVENT_CONFIRMED,
                                   "FAULT_" + std::to_string(i), i));
  }

  auto req = make_stream_request("127.0.0.1", "1");
  httplib::Response res;
  handler_->handle_stream(req, res);

  auto output = read_stream_once(res, 1);
  EXPECT_NE(output.find("id: 2\n"), std::string::npos);
  EXPECT_EQ(parse_sse_payload(output)["fault"]["fault_code"], "FAULT_2");

  release_stream(res);
}
```

File: src/ros2_medkit_gateway/include/ros2_medkit_gateway/http/handlers/sse_fault_handler.hpp [hunk 4]
```diff
@@ -56,6 +57,8 @@ class SSEFaultHandler {
    * @param client_tracker Shared SSE client counter (across all SSE handlers)
    */
   SSEFaultHandler(HandlerContext & ctx, std::shared_ptr<SSEClientTracker> client_tracker);
+  SSEFaultHandler(HandlerContext & ctx, std::shared_ptr<SSEClientTracker> client_tracker,
```
Comment (bburda) [inline src/ros2_medkit_gateway/include/ros2_medkit_gateway/http/handlers/sse_fault_handler.hpp:66]: **[Important]** This test-only constructor is public with no documentation. The existing 2-arg constructor above has `@brief` and `@param` annotations. Without a doc comment, readers cannot tell this is a test seam, and any consumer of `gateway_lib` (including future plugins) can construct an `SSEFaultHandler` with a 1ms keepalive in production.

At minimum add a Doxygen block:
```cpp
/// @brief Test-only constructor that overrides the keepalive interval.
/// @param keepalive_interval Must be positive; non-positive values fall back to the 30s default.
```
Alternatively, make it `private` with `friend class SSEFaultHandlerTest;`.

File: src/ros2_medkit_gateway/src/http/handlers/sse_fault_handler.cpp [hunk 5]
```diff
@@ -27,7 +27,16 @@ namespace ros2_medkit_gateway {
 namespace handlers {
 
 SSEFaultHandler::SSEFaultHandler(HandlerContext & ctx, std::shared_ptr<SSEClientTracker> client_tracker)
-  : ctx_(ctx), client_tracker_(std::move(client_tracker)) {
+  : SSEFaultHandler(ctx, std::move(client_tracker), std::chrono::seconds(kKeepaliveIntervalSec)) {
+}
+
+SSEFaultHandler::SSEFaultHandler(HandlerContext & ctx, std::shared_ptr<SSEClientTracker> client_tracker,
+                                 std::chrono::milliseconds keepalive_interval)
+  : ctx_(ctx)
```
Comment (bburda) [inline src/ros2_medkit_gateway/src/http/handlers/sse_fault_handler.cpp:47]: **[Important]** Pre-existing issue (not introduced by this PR): `event.event_type` from the ROS 2 message is written verbatim into the SSE `event:` line in `format_sse_event` without newline sanitization. SSE framing is newline-delimited, so an `event_type` containing `\n` could inject arbitrary SSE events into connected clients' streams. Currently mitigated by fixed constants in `FaultManagerNode`, but exploitable by a rogue node publishing directly to `~/events`.

Since the new tests exercise this path, consider adding a sanitization guard in `format_sse_event` - strip `\r`/`\n` from `event_type` before formatting.

File: src/ros2_medkit_gateway/src/http/handlers/sse_fault_handler.cpp [hunk 6]
```diff
@@ -27,7 +27,16 @@ namespace ros2_medkit_gateway {
...
+  , client_tracker_(std::move(client_tracker))
```
Comment (bburda) [inline src/ros2_medkit_gateway/src/http/handlers/sse_fault_handler.cpp:48]: **[Important]** Non-positive `keepalive_interval` silently falls back to 30s with no warning. If a test accidentally passes `0ms`, it will block for 30 seconds with no indication why - making debugging painful.

Consider `assert(keepalive_interval > std::chrono::milliseconds::zero())` for fast-fail in tests, or at minimum emit `RCLCPP_WARN` when the override is rejected.
